battle_round.go

```diff
@@ -15,6 +17,9 @@ func (b *Battle) preRound() {
 				StatusEffect: StatusSleep,
 			})
 		}
+		if pkmn.Ability == AbilityAirLock {
+			b.metadata[MetaWeatherDisabled] = true
+		}
```

Comment:
Should this be another transaction so it's possible to announce it? Something like `AbilityEffectTransaction`.

Reply:
I think it would be hard to make a transaction generic enough to be able to handle these types of cases. Right now the metadata fields are the only `interface{}` types. I also considered this being a potential battle rule (to disable weather), which doesn't make sense for it being a transaction.

This seems like the easiest way to ensure that we always have this check in the beginning of a round.